lib/posix/pthread.c

```diff
@@ -636,29 +635,42 @@ int pthread_join(pthread_t pthread, void **status)
 	}
 
 	ret = 0;
-	key = k_spin_lock(&pthread_pool_lock);
-	if (t->detachstate != PTHREAD_CREATE_JOINABLE) {
-		ret = EINVAL;
-	} else if (t->qid == POSIX_THREAD_READY_Q) {
-		/* marginal chance thread has moved to ready_q between to_posix_thread() and here */
-		ret = ESRCH;
-	} else {
+	K_SPINLOCK(&pthread_pool_lock)
+	{
+		if (t->detachstate != PTHREAD_CREATE_JOINABLE) {
+			ret = EINVAL;
+			K_SPINLOCK_BREAK;
+		} else if (t->qid == POSIX_THREAD_READY_Q) {
```

Comment:
minor comment here -- you're using `K_SPINLOCK_BREAK` in the `if` clause above, so you don't actually need the `else` here. Any idea what conventions with `K_SPINLOCK_BREAK` are these days? And, thanks for exposing us to K_SPINLOCK. That's a sweet macro.

Reply:
Fixed!